lorax/lorax.py

```diff
@@ -33,24 +34,340 @@ class TheLorax(object):
+        # We need to flip the sign of the scores.
+        if pred_class == 0:
+            score = 1.0 - score
+            contrib_list = [(feature, score * -1) for feature, score in contrib_list]   
+
+        # TODO: Need to be modified to not taking the index
+        # Replacing the example id with -1 for now
+        logging.info('Used predicted class {} for example {}, score={}'.format(pred_class,
+                                                                            -1,
+                                                                            score))
+
+        # sorting in descending order by contribution then by feature name in the case of ties
+        contrib_list.sort(key=lambda x: (x[1] * -1, x[0]))
+
+        # TODO: If descriptive is set, the importance scores
+        # are supported with the context provided by a test dataset
+        # The code is available in the original constructor, move it here
+        if descriptive:  
+            # If descriptive, it rolls back to the original case         
+            if test_mat is None:
+                test_mat = self.X_test
+                fstats = self.feature_stats
+            else:
+                fstats = self.populate_feature_stats(test_mat)
+
+            contrib_df = self._build_contrib_df(contrib_list,
+                                                test_mat=test_mat,
+                                                idx=idx,
+                                                sample=sample,
+                                                feature_stats=fstats,
+                                                how=how)
+
+            # adding overall feature importance from model level
+            overall_importance = []
+            for i, cname in enumerate(self.column_names):
+                if isinstance(self.clf, LogisticRegression):
+                    overall_importance.append((cname, self.clf.coef_[0][i]))
+                
+                elif isinstance(self.clf, RandomForestClassifier):
+                    overall_importance.append((cname, self.clf.feature_importances_[i]))
+
+                else:
+                    pass
+
+            updated_list = add_overall_feature_importance(contrib_list, overall_importance)
+            updated_columns = ['feature', 'sample_rank', 'overall_imp', 'overall_rank', 'rank_change']
+            contrib_df = contrib_df.join(
+                pd.DataFrame(
+                    data=updated_list, 
+                    columns=updated_columns
+                ).set_index('feature')
+            )
+
+            if graph:  
+                self._plot_graph(idx,
+                                 pred_class,
+                                 score,
+                                 num_features,
+                                 contrib_df,
+                                 how)
+
+        else:
+            contrib_df = self._build_contrib_df_sample(contrib_list, how=how)
+
+        return contrib_df
```

Comment:
Also not necessary for this PR, but I wonder if we should somehow try to return the matplotlib axis/figure object as well when graphing. Might make sense for an issue to look at in the future, though.